src/options.ts

```diff
@@ -111,9 +113,29 @@ export function _setDefaultOptions(options: Partial<Options> = {}): Options {
     ...extraTracerConfig,
   };
 
+  if (options.tracesExporter === undefined) {
+    options.tracesExporter = (process.env.OTEL_TRACES_EXPORTER ||
+      TracesExporter.OTLP) as TracesExporter;
+  }
+
   // factories
-  options.spanExporterFactory =
-    options.spanExporterFactory || defaultSpanExporterFactory;
+  if (options.spanExporterFactory === undefined) {
+    if (options.tracesExporter === TracesExporter.JAEGER) {
+      options.spanExporterFactory = jaegerSpanExporterFactory;
+      options.endpoint =
+        options.endpoint ||
+        otelEnv.OTEL_EXPORTER_JAEGER_ENDPOINT ||
+        'http://localhost:14268/v1/traces';
```

Comment:
options are passed as an argument to spanExporterFactory. Each factory should host code to parse/set defaults internally.

Reply:
To clarify, I mean this function should not have specific knowledge about each exporter. it should just pass on the options instance to the exporter factory. Then exporter factory should read it's relevant env vars, options.endpoint, etc.

Reply:
Rest of `_setDefaultOptions` sets and parses defaults for other parts of the API and we already pass the factories the whole `options` object instead of letting it take some of it's configuration from `options` and rest from the env. I don't see how exporters are different.

Reply:
What I meant was that the logic for configuring each exporter is leaking outside the factory to the constructor. I think constructor should do as little as possible aand only try to identify the components to be used. Then let the components configure themselves, including handling their own defaults, validation logic etc.

Reply:
Split the factories up.